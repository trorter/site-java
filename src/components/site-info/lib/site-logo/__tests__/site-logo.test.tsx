import {render, screen} from "@testing-library/react";
import renderer, {act} from "react-test-renderer";
import SiteLogo from "../site-logo";
import {MemoryRouter} from "react-router-dom";
import React, {ReactElement} from "react";
import {Simulate} from "react-dom/test-utils";
import click = Simulate.click;

export {}

const routerRender = (ui: ReactElement) => {
    return render(ui, {wrapper: MemoryRouter});
}

const routerRenderer = (ui: ReactElement) => {
    return renderer.create(
        <MemoryRouter>
            {ui}
        </MemoryRouter>);
}

describe("Should render", () => {
    describe("With props", () => {

        it("and pass the snapshot test", () => {
            const rendererResult = routerRenderer(<SiteLogo/>).toJSON();
            expect(rendererResult).toMatchSnapshot();
        })

        it("with default ones", async () => {
            routerRender(<SiteLogo/>);

            expect(await screen.findByTestId("SiteLogoTest"))
                .toBeInTheDocument();

            expect(await screen.findByAltText(SiteLogo.defaultProps?.alt || ""))
                .toBeInTheDocument();

            expect(await screen.findByTestId("SiteLogoTest"))
                .toMatchObject(/images\/main_icon.svg/);

            expect(await screen.findByTestId("SiteLogoTest"))
                .not
                .toHaveAttribute('preview');

        })

        it("with set alt", async () => {
            const alt = "alt";

            routerRender(<SiteLogo alt={alt}/>);

            expect(await screen.findByAltText(alt))
                .toBeInTheDocument();
        })

        it("with set preview", async () => {
            const preview = true;

            routerRender(<SiteLogo preview={preview}/>);

            expect(await screen.findByTestId("SiteLogoTest"))
                .toBeInTheDocument();

        })
    })

    describe("with click", () => {
        it("on icon", async () => {
            routerRender(<SiteLogo />);

            const icon = await screen.findByTestId("SiteLogoTest");
            act(() => {
                click(icon);
            })

            expect(window.location.pathname).toBe("/");
        })
    })
});