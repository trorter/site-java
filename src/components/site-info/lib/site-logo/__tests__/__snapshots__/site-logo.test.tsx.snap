// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render With props and pass the snapshot test 1`] = `
<div
  className="ant-image css-dev-only-do-not-override-yp8pcc"
  data-testid="SiteLogoTest"
  onClick={[Function]}
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <img
    alt="â‡ˆ"
    className="ant-image-img"
    onLoad={[Function]}
    src="/images/main_icon.svg"
    style={
      Object {
        "height": 52,
        "marginBottom": 2,
      }
    }
  />
</div>
`;
